spec/rspec/core/memoized_helpers_spec.rb

```diff
@@ -363,6 +363,97 @@ def not_ok?; false; end
         expect(subject).to eq(3)
       end
     end
+
+    describe 'threadsafety', :threadsafe => true do
+      # Force call to eq to happen in threaded env to prevent "circular require" warnings
+      # Not actually sure why calling eq in two threads at once causes this (it's a sporadic failure):
```

Comment:
I can answer this...matchers are autoloaded:

https://github.com/rspec/rspec-expectations/blob/v3.2.0/lib/rspec/matchers/built_in.rb

So the first time a matcher is used it loads the class definition.  That's probably what's happening here.  Although, from what I've read, recent rubies (e.g. 2.0+, I think?) have made `autoload` threadsafe.